ouroboros-consensus-test/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -77,6 +92,217 @@ tests = testGroup "Mempool"
   , testProperty "removeTxs"                               prop_Mempool_removeTxs
   ]
 
+{-------------------------------------------------------------------------------
+  Mempool Pure Properties
+-------------------------------------------------------------------------------}
+
+-- | Test that removed transactions are not in the
+--   result transactionsIds of the returned internal state.
+prop_pure_removeTxs
+  :: TestSetupSub
+  -> Property
+prop_pure_removeTxs TestSetupSub {..} =
+    withInternalState tssSetup $
+    \ mpArgs internalSt ledgerState ->
+      let (internalStRes, _)    = pureRemoveTxs (map txId tssSubTxs) mpArgs internalSt ledgerState
+          txIdsRemaining        = Set.toList (isTxIds internalStRes)
+      in property $ null $ intersect txIdsRemaining (map txId tssSubTxs)
+
+-- | Test that after removing transactions all the other transactions are still available.
+--
+prop_pure_removeTxs_contra
+    :: TestSetupWithTxs
+    -> Property
+prop_pure_removeTxs_contra TestSetupWithTxs {..} =
+    withInternalState testSetup $
+    \ mpArgs internalSt ledgerState ->
+      let (_res,internalSt')     = runTryAddTxs mpArgs internalSt (map fst txs)
+          (internalStRes, _tra)  = pureRemoveTxs (map (txId . fst) txs)
+                                      mpArgs internalSt' ledgerState
+          txIdsRemaining         = Set.toList (isTxIds internalStRes)
```

Comment:
If the arguments of a function call don't fit on a line, we put each of them on a separate line.

```suggestion
      let (_res, internalSt')   = runTryAddTxs mpArgs internalSt (map fst txs)
          (internalStRes, _tra) = pureRemoveTxs
                                    (map (txId . fst) txs)
                                    mpArgs
                                    internalSt'
                                    ledgerState
          txIdsRemaining        = Set.toList (isTxIds internalStRes)
```